### ISSUE TITLE
progress bar display error when calling model.evaluate method after a load_model or clone_model

### ISSUE THREAD

**Issue by franckma31** [tensorflow/tensorflow#51989]
progress bar display error when calling model.evaluate method after a load_model or clone_model

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): 
Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 
Ubuntu 20.04.3 LTS (GNU/Linux 5.4.0-80-generic x86_64)
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
No
- TensorFlow installed from (source or binary):
Binary (pip install tensorflow==2.4)
- TensorFlow version (use command below):
v2.4.0-rc4-71-g582c8d236cb 2.4.0
- Python version:
Python 3.8.1
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
cuda-11.2
cudnn/11.0-v8.0.0
- GPU model and memory:
GeForce RTX 2070

**Describe the current behavior**
After a clone_model, or a load_model, the model.evaluate method doesn't display the same metrics results in progress_bar and in returned values (wrong ones in progress bar):

results = model_copy.evaluate(test_ds2)
print(results)

40/40 [==============================] - 0s 1ms/step - loss: 0.1934 - accuracy: **0.9419**
[0.16363045573234558, **0.9517999887466431**]

It seems that ProgbarLogger  stateful_metrics is not set for loss and metrics (here accuracy), due to a late assignment to self.model.metrics (tensorflow/python/keras/callbacks.py l.1047) leading to an average value display of these values instead of a single final value (tensorflow/python/keras/utils/generic_utils.py l.551)

**Describe the expected behavior**
This behavior is different when the evaluate function is called after a fit call (which set the model.metrics) and lead to the right display
40/40 [==============================] - 0s 1ms/step - loss: 0.1636 - acc: **0.9518**
[0.16363045573234558, **0.9517999887466431**]

**[Contributing](https://www.tensorflow.org/community/contribute)**

- Do you want to contribute a PR? (yes/no):
NO, not sure to see where to handle this issue
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

few lines codes with MNIST and simplest model.

```
import tensorflow as tf
from tensorflow.keras import backend as K
from tensorflow.keras.layers import ReLU
from tensorflow.keras.optimizers import Adam
from tensorflow.keras import layers


physical_devices = tf.config.experimental.list_physical_devices('GPU')
tf.config.experimental.set_memory_growth(physical_devices[0], True)


from tensorflow.keras.datasets import mnist
(x_train, y_train), (x_test, y_test) = mnist.load_data()


def prepare_data(x, y):
    x=x.astype('float32')
    y=y.astype('int32')
    # convert from range int[0,255] to float32[-1,1]
    x/=255
    x = 2*x -1
    x=x.reshape((-1,28,28,1))
    y=tf.keras.utils.to_categorical(y,num_classes=10)
    return x, y

# prepare the data
x_train, y_train = prepare_data(x_train, y_train)
x_test, y_test = prepare_data(x_test, y_test)


epochs = 2 #200
batch_size = 256

##simplest model
K.clear_session()
model = tf.keras.Sequential([
    layers.Flatten(),
    layers.Dense(100),
    layers.ReLU(),
    layers.Dense(120),
    layers.ReLU(),
    layers.Dense(10)
]
)

loss_function = tf.losses.CategoricalCrossentropy(from_logits=True)
optimizer = Adam(lr=0.001)
model.compile(loss=loss_function,optimizer=optimizer, metrics=['acc'])


model.fit(x_train,y_train,batch_size=batch_size,epochs=epochs,validation_data=(x_test,y_test))


print("Run evaluate on test dataset")
test_ds = tf.data.Dataset.from_tensor_slices((x_test, y_test)).batch(batch_size)
results = model.evaluate(test_ds)
print(results)

##---------------------------
print("Cpy model ")
model_copy= tf.keras.models.clone_model(model)
model_copy.build((None, 28,28,1)) # replace 10 with number of variables in input layer
model_copy.compile(loss=loss_function,optimizer=optimizer, metrics=["accuracy"])
model_copy.set_weights(model.get_weights())

print("Run evaluate with dataset on copied model")
test_ds2 = tf.data.Dataset.from_tensor_slices((x_test, y_test)).batch(batch_size)
results = model_copy.evaluate(test_ds2)
print(results)  

```
**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

**Comment by tilakrayal** [comment]
@franckma31 ,
I was able to run the mentioned code and get the same result in tf v2.6. Could you please create a virtual environment and test your code again. It helps. Thanks!

**Comment by franckma31** [comment]
Hi Tilakrayal, 
Thanks for your answer.
Here is a colab notebook that should reproduce the bug with tf==2.4

[https://colab.research.google.com/drive/19_vQm5LeR82PiT7uZd20WyEtKo652D3p?usp=sharing](url)

**Comment by tilakrayal** [comment]
@franckma31 ,
Please upgrade to latest stable tensorflow v2.6 which resolves the issue.Thanks!

**Comment by franckma31** [comment]
Yes I know that this issue is resolved in 2.5 and 2.6, but I do think that I'm not the only tensorflow user that, due to other lib dependencies, cannot upgrade to 2.5. 
And that means that tf users will report false accuracy just because they rely on the progbar information. TF 2.4 is only 9 months old, I thought that it was still maintained, and that this issue could help. Quite a bit disapointed.

If for you this is not an issue, please feel free to close it

**Comment by tilakrayal** [comment]
@franckma31 ,
As it was bug in tf v2.4, it has been resolved in latest stable version 2.6.Please upgrade to v2.6 and try to test your code.It helps.Thanks

**Comment by franckma31** [comment]
No comment. As I said even if on this simple example, you can verify on the colab that it does work on 2.6, for my project I can not upgrade to 2.6 and I do think that  such a bug should be fixed even for 2.4, it may lead to false results in publications, or industrial products. Regards

**Comment by Saduf2019** [comment]
@franckma31 
As there have been below changes and the issue is resolved in tf 2.6, upgrading would help as many bugs get fixed in later versions, in case is still remains an issue please open this in keras and move this issue to closed status.

Please post this issue on [keras-team/keras repo](https://github.com/keras-team/keras/issues).
To know more refer to:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
